[Draft]Add Multimodal RAG notebook

openvino-dev-samples opened the pull request:
![image](https://github.com/user-attachments/assets/a8ebf3fc-7a34-416b-b744-609965792744)

![image](https://github.com/user-attachments/assets/b6f97e32-d567-4278-afac-a633776b463d)

openvino-dev-samples committed:
first draft

openvino-dev-samples committed:
reformat

openvino-dev-samples committed:
fix ci

openvino-dev-samples committed:
add gradio demo

openvino-dev-samples committed:
reformat

reformat

openvino-dev-samples committed:
update gradio UI

eaidova commented on notebooks/multimodal-rag/multimodal-rag-llamaindex.ipynb:
```diff
@@ -0,0 +1,1054 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #12.        convert_phi3_model(vlm_model_id, vlm_model_path, compression_configuration)
</pre><p><a href="https://github.com/huggingface/optimum-intel/pull/977" target="_blank">https://github.com/huggingface/optimum-intel/pull/977</a> phi3-vision is planned to be integrated to optimum-intel</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2497/discussion/#comment-1830717253'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='9'/>

openvino-dev-samples replied:
Done, replace with optimum-intel

openvino-dev-samples committed:
transfer to optimum-intel

transfer to optimum-intel

transfer to optimum-intel

openvino-dev-samples committed:
Merge branch 'latest' into multimodal-rag

openvino-dev-samples committed:
fix spelling

eaidova commented on notebooks/multimodal-rag/multimodal-rag-llamaindex.ipynb:
```diff
@@ -0,0 +1,1168 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #3.        "transformers==4.45" \
</pre><p>what the reason for pinning transforemrs version?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2497/discussion/#comment-1846060949'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='2'/>

aleksandr-mokrov replied:
<pre class="ql-syntax" spellcheck="false">Line #7.    demo = make_demo(filepath, build_index, search, generate, load_image, stop)
</pre><p>There are no <code>load_image </code>argument in <code>make_demo </code>function.﻿</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2497/discussion/#comment-1846792334'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='40'/>

eaidova commented:
@openvino-dev-samples for me everything looks good, thanks.

Couple of comments:
Possibly it is better to move on accuracy aware quantization for VLM using optimum-cli, need to provide `--weight-format int4 --dataset contextual` options for that (fyi @nikita-savelyevv)

Is there any plans to integrate OV Visual Language models directly in llama-index?

openvino-dev-samples commented:
> @openvino-dev-samples for me everything looks good, thanks.
> 
> Couple of comments: Possibly it is better to move on accuracy aware quantization for VLM using optimum-cli, need to provide `--weight-format int4 --dataset contextual` options for that (fyi @nikita-savelyevv)
> 
> Is there any plans to integrate OV Visual Language models directly in llama-index?

Thanks for your review, the integration is already done in llama-index 

https://docs.llamaindex.ai/en/stable/examples/multi_modal/openvino_multimodal/

BTW is there an example for phi3-vision's accuracy aware quantization ?

nikita-savelyevv commented:
> Couple of comments: Possibly it is better to move on accuracy aware quantization for VLM using optimum-cli, need to provide `--weight-format int4 --dataset contextual` options for that (fyi @nikita-savelyevv)


I'll add that an algorithm itself needs to be specified, e.g. `--weight-format int4 --dataset contextual --awq`.

Also, the default number of samples of 128 might be too large, so it can be reduced with `--num-samples 32`.

openvino-dev-samples committed:
add load image function

add load image function

openvino-dev-samples commented:
> > Couple of comments: Possibly it is better to move on accuracy aware quantization for VLM using optimum-cli, need to provide `--weight-format int4 --dataset contextual` options for that (fyi @nikita-savelyevv)
> 
> I'll add that an algorithm itself needs to be specified, e.g. `--weight-format int4 --dataset contextual --awq`.
> 
> Also, the default number of samples of 128 might be too large, so it can be reduced with `--num-samples 32`.

Hi, as my test, the accuracy with this configuration is not satisfied:
`optimum-cli export openvino --model {vlm_model_id}  {vlm_model_path} --trust-remote-code --weight-format int4 --dataset contextual --awq --num-samples 32`

nikita-savelyevv commented:
> > > Couple of comments: Possibly it is better to move on accuracy aware quantization for VLM using optimum-cli, need to provide `--weight-format int4 --dataset contextual` options for that (fyi @nikita-savelyevv)
> > 
> > 
> > I'll add that an algorithm itself needs to be specified, e.g. `--weight-format int4 --dataset contextual --awq`.
> > Also, the default number of samples of 128 might be too large, so it can be reduced with `--num-samples 32`.
> 
> Hi, as my test, the accuracy with this configuration is not satisfied: `optimum-cli export openvino --model {vlm_model_id} {vlm_model_path} --trust-remote-code --weight-format int4 --dataset contextual --awq --num-samples 32`

Thanks for the information! Have you compared it against the configuration below?
```
compression_config = {
    "mode": nncf.CompressWeightsMode.INT4_SYM,
    "group_size": 64,
    "ratio": 0.6,
}
```

Yes, this configuration brings more reasonable responses compared to optimum-cli

openvino-dev-samples committed:
update the method of audio extraction

update the method of audio extraction

openvino-dev-samples committed:
replaace video upload component

openvino-dev-samples committed:
update the screenshot display method

update the screenshot display method

openvino-dev-samples commented:
hi @eaidova looks the CI is out of resource to validate this notebook, any suggestions ? thanks

eaidova commented:
@openvino-dev-samples I can suggest to try internvl2-1b-instruct or nano-llava for ci testing, it is small enough to fit into precommit

openvino-dev-samples committed:
replace llm test case

fix ci issues

openvino-dev-samples committed:
update with accruaracy aware quantization

update with accruaracy aware quantization

openvino-dev-samples commented:
> --trust-remote-code --weight-format int4 --dataset contextual --awq --num-samples 32

Sorry, I made a mistake before. The result looks good with this accuracy aware config now

openvino-dev-samples committed:
solve conflict

openvino-dev-samples committed:
solve conflict

solve conflict

openvino-dev-samples committed:
switch to int8 ASR

openvino-dev-samples commented:
> > --trust-remote-code --weight-format int4 --dataset contextual --awq --num-samples 32
> 
> Sorry, I made a mistake before. The result looks good with this accuracy aware config now

However I find it impossible to run this quantization method in a client PC with 32GB RAM

openvino-dev-samples committed:
skip macos

nikita-savelyevv commented:
> > > --trust-remote-code --weight-format int4 --dataset contextual --awq --num-samples 32
> > 
> > 
> > Sorry, I made a mistake before. The result looks good with this accuracy aware config now
> 
> However I find it impossible to run this quantization method in a client PC with 32GB RAM

Could you please verify that NNCF 2.14 is installed? It was released recently and it contains significant improvements in terms of peak RAM during data-aware compression.

brmarkus commented:
> Could you please verify that NNCF 2.14 is installed? It was released recently and it contains significant improvements in terms of peak RAM during data-aware compression.

At least this pull-request doesn't touch "requirements.txt" and doesn't check to make sure a recent version of NNCF is present. On my 64GB laptop the max. system memory is shortly fully used.

openvino-dev-samples committed:
solve conflict

openvino-dev-samples committed:
reduce the number of frame saved

openvino-dev-samples committed:
solve conflict

openvino-dev-samples committed:
add skipped os

eaidova committed:
Merge branch 'latest' into multimodal-rag

openvino-dev-samples committed:
update video url

openvino-dev-samples committed:
change the ASR model id

change the ASR model id

change the ASR model id

change the ASR model id

change the ASR model id

change the ASR model id

change the ASR model id

change the ASR model id

change the ASR model id

openvino-dev-samples committed:
update ov version

openvino-dev-samples committed:
ignore mul-rag in docker ci

openvino-dev-samples committed:
Merge branch 'latest' into multimodal-rag
